sleap_nn/training/model_trainer.py

```diff
@@ -313,16 +457,30 @@ def train(self):
                     wandb_logger.experiment.config.update({key: value})
                 wandb_logger.experiment.config.update({"model_params": total_params})
 
-        except Exception:
+        except KeyboardInterrupt:
             print("Stopping training...")
 
         finally:
             if self.config.trainer_config.use_wandb:
                 self.config.trainer_config.wandb.run_id = wandb.run.id
-                self.config.model_config.total_params = total_params
-                wandb.finish(exit_code=255)
+                wandb.finish()
+            self.config.model_config.total_params = total_params
             # save the configs as yaml in the checkpoint dir
-            OmegaConf.save(config=self.config, f=f"{dir_path}/training_config.yaml")
+            OmegaConf.save(
+                config=self.config, f=f"{self.dir_path}/training_config.yaml"
+            )
```

Comment:
Should happen _before_ model training so we have the computed config values in the YAML file for troubleshooting if the training fails.

Before L442 would be good.